### ISSUE TITLE
Change to Markdown

### ISSUE THREAD

**Issue by chaudhuri** [abella-prover/abella-prover.org#2]
These web-pages should be changed to use Markdown instead.
They should also be generated from a static site generator such as [Hakyll](https://jaspervdj.be/hakyll/) or [STOG](https://zoggy.github.io/stog/).

**Comment by amerikan** [comment]
Maybe Github Pages can be considered: https://pages.github.com/

**Comment by chaudhuri** [comment]
I considered it and rejected it in the past since github pages don't play nicely with arbitrary domains. They want to take full control over the root CNAME.

Anyway it's not the hard part. Any git push to master on this repo is automatically mirrored on the Abella web-site with a commit hook. The bigger issue is to convert a lot of hand-coded HTML to Markdown in such a way that a Markdown processor like pandoc gives us back something like the original HTML page. This is a bit difficult because we use custom AJAX for showing/hiding proof steps.

Maybe we should do this in two stages: the static pages first, and then the example pages later (if at all).

**Comment by amerikan** [comment]
Is AJAX only being used for the purpose of showing/hiding content? If so, there  are _CSS_ only methods to accomplish such functionality.

**Comment by chaudhuri** [comment]
Actually, I just looked and there is no AJAX at all. I think I meant to say just Javascript there.

Showing/hiding cannot be done with CSS alone, I believe. You will need at least some Javascript to set the proper CSS properties to show/hide divs.

Anyhow, the overall issue is that github pages solves a problem we don't have. Every commit to master on this repository is immediately mirrored on `abella-prover.org` using a commit hook. Moreover, Github Pages is not a static website "compiler" like Hakyll.

**Comment by amerikan** [comment]
Are you sure javascript usage is not allowed in markdown?

**Comment by chaudhuri** [comment]
No, I just meant that the Abella web server runs completely hands off and simply mirrors the git repository. It in fact runs in a container that has next to nothing but git installed. I'll take a look at building a container with a markdown compiler so that the commit hook can still work as it does today.

GitHub pages are looking like a good option these days since they have gotten a lot easier to use on custom domains. Maybe it's worth taking another look at them after the conference deadline season.

**Comment by amerikan** [comment]
Yes it's quite easy if you want to give the github pages a subdomain. You just have to enable github pages and some DNS changes.

The DNS would entry would be something like:

`CNAME` `docs.abella-prover.org` -> `abella-prover.github.io.`

You'd have to renamed this repo as `docs` though since it should match the subdomain you used  in the CNAME. Of course that is just an example as you can call it whatever you want.
